main/src/main/java/com/expedia/metrics/HashIdFactory.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2018 Expedia Group, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expedia.metrics;
+
+import javax.xml.bind.DatatypeConverter;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+public class HashIdFactory implements IdFactory {
+
+    DefaultIdFactory defaultIdFactory;
+
+    public HashIdFactory() {
+        defaultIdFactory = new DefaultIdFactory();
+    }
+
+    @Override
+    public String getId(MetricDefinition metric) {
+        MessageDigest md;
+        try {
+            md = MessageDigest.getInstance("MD5");
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException("The JRE is missing required digest algorithm MD5", e);
+        }
+        md.update(defaultIdFactory.getId(metric).getBytes(StandardCharsets.UTF_8));
+
+        byte[] digest = md.digest();
+        return DatatypeConverter.printHexBinary(digest).toLowerCase();
```

Comment:
Use `Encoder.encodeHex` for this. See `MetricKey.toString()` for an example of usage.